### ISSUE TITLE
Pass numbers through convertNumber

### ISSUE THREAD

**Issue by Huji** [azatoth/twinkle#987]
When a number is dynamically added to an interface message, pass it through `mw.language.convertNumber()` first.

In English, it helps by adding commas far large numbers. For example, in the batchdelete module, the size of a fill is [currently](https://github.com/azatoth/twinkle/blob/d3e40f37457f3fdb42fc2340d543b15d0a4ab8cc/modules/twinklebatchdelete.js#L191) inserted into a message using `metadata.push(size + ' bytes');` and changing it to `metadata.push(mw.language.convertNumber(size) + ' bytes');` will result in "143908 bytes" to read as "143,908 bytes" instead.

It also helps with #129 for non-English languages (e.g. those that use digits other than 123, or those that use a different "thousands separator")

**Comment by Amorymeltzer** [comment]
I guess that'd be helpful?  I dunno, their uses in the `batch`es are mostly a "is there appreciable content or not" so the number is less valuable, but I suppose have the appropriate places might make it easier.

I've actually been thinking of using `mediawiki.language` a bit.  I had thought it was an additional requirement to load, but I guess that's just `.months` and the like?

**Comment by Huji** [comment]
I think `mediawiki.language` is another dependency, but an important one, especially if we really intent to address #129

**Comment by siddharthvp** [comment]
it's another dependency in that it needs to be specified in the gadget definition (check mw.loader.moduleRegistry). But it's loaded by default so it doesn't add any extra load. 

But we should be cognizant of the frustrating limitation that mw.language non-negotiably uses the wgUserLanguage rather than wgContentLanguage. In the PR, I see that convertNumber is only used for interface messages rather than in editing. So we're okay here (though a person who has set his interface language as fa would need to brace for some surprises when they use TW on enwiki).

The language data it uses comes from mw.language.data - which is exported from the backend only for the user language (by [ResourceLoaderLanguageDataModule.php](https://doc.wikimedia.org/mediawiki-core/master/php/ResourceLoaderLanguageDataModule_8php_source.html)). The corresponding data for the wgContentLanguage exists on the server, of course, but there's no way to get to it, even via the API. One thing that would be helpful is to create a new API that exposes the language data to begin with, though it's still a problem that mw.language is configured to only look at the user language data.

An even better solution to solve TW's i18n problems would be to add a server-side component to Twinkle (aka convert it to an extension), so that we can read data from the server and export whatever we need.

**Comment by Huji** [comment]
I don't think we should try to support a situation where a user chooses a different interface language than that content language of the wiki yet. Maybe in the future, but not now; and certainly not until #129 is resolved. That scenario would mean we would want the Twinkle interface to be in user's language, but the edit summaries, etc. to be in content language (for the sake of other users) and handling this duality is nontrivial.

What is suggested here (the use of `convertNumber`) is needed even for the simpler case of wiki language = user language but both != 'en'
